### DISCUSSION TITLE
How to estimate what cost my Artillery load test will have on my AWS bill?

### DISCUSSION THREAD

**Discussion by SonnyFishback** [artilleryio/artillery#2692] [Artillery]
I see that Artillery provides an estimate of how much a test would cost after the test runs. How can I estimate how much a test will cost as I plan my test?

I would be using AWS Fargate to run a Playwright test of 60K virtual users with 3 minute sessions over at least a one hour period.

In the future we want to test for Black Friday traffic which is about 1.2 million users in a 24 hour period.

**Comment by hassy** [comment]
hi @SonnyFishback 👋 Detailed AWS cost estimation can be challenging to say the least. For the load test you're describing the main factors would be:

1. Fargate compute cost. This is a straightforward calculation and most likely to be the main driver of the cost. The variables here are CPU/memory configuration for the containers in the test, multiplied by number of containers and duration.
2. Outbound bandwidth cost - this is more complicated and depends on the amount of data your tests will be sending out (e.g. are large file uploads involved), and can also depend on the exact networking configuration in your AWS account.
3. Cost of any other services your test scripts might use, some common ones would be accessing Dynamo or Elasticache, or looking up secrets in AWS Parameter Store.

Happy to discuss in more detail, feel free to email me on hveldstra@artillery.io with more details that might not be suitable for a public discussion.

**Reply by SonnyFishback** [reply]
Thank you for the swift response! Sending you an email. @hassy
